### PR TITLE
[DOCS] EQL: Add `:` operator, remove wildcard operator

### PR DESCRIPTION
**Pull request by jrodewig**
Documents the case-insensitive `:` operator. Removes docs for the wildcard (`*`) matching operator.

Relates to #63121 and #63262

### Preview
https://elasticsearch_63195.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/eql-syntax.html#eql-syntax-conditions

### CODE REVIEW

File: docs/reference/eql/syntax.asciidoc [hunk 1]
```diff
@@ -100,14 +100,19 @@ to the right. Otherwise returns `false`.
...
+`==` (equal, case-insensitive)::
 Returns `true` if the values to the left and right of the operator are equal.
-Otherwise returns `false`.
-+
+Otherwise returns `false`. For strings, matching is case-insensitive.
+
+`:` (equal, case-sensitive)::
+Returns `true` if strings to the left and right of the operator are equal.
+Otherwise returns `false`. Matching is case-sensitive and can only be used to
+compare strings. Wildcard (`*`) patterns are not supported.
```
Comment (jrodewig) [inline docs/reference/eql/syntax.asciidoc:110]: > Wildcard (`*`) patterns are not supported.

This isn't technically the case yet, but I included this as that's the current implementation plan.

**Comment by jrodewig** [comment]
Now that #63218 is merged, I'll open a separate PR to document those changes.

File: docs/reference/eql/syntax.asciidoc [hunk 2]
```diff
@@ -127,14 +127,19 @@ to the right. Otherwise returns `false`.
...
+`==` (equal, case-sensitive)::
 Returns `true` if the values to the left and right of the operator are equal.
-Otherwise returns `false`.
-+
+Otherwise returns `false`. For strings, matching is case-sensitive.
+
+`:` (equal, case-insensitive)::
+Returns `true` if strings to the left and right of the operator are equal.
+Otherwise returns `false`. Matching is case-insensitive and can only be used to
+compare strings. Wildcard (`*`) patterns are not supported.
```
Comment (jrodewig) [inline docs/reference/eql/syntax.asciidoc:137]: > Wildcard (`*`) patterns are not supported.

It looks like the operator currently supports these. Let me know if this should be removed.

File: docs/reference/eql/syntax.asciidoc [hunk 3]
```diff
@@ -369,15 +376,17 @@ use a regular string with the `\"` escape sequence.
...
+When comparing strings using the `==` or `!=` operators, you can use the
+wildcard (`*`) operator within the string to match specific patterns:
 
 [source,eql]
 ----
 field == "example*wildcard"
 field != "example*wildcard"
 ----
 
+The `:` operator does not support wildcard patterns.
```
Comment (jrodewig) [inline docs/reference/eql/syntax.asciidoc:388]: Similar to above note. Currently, the `:` operator does support wildcards.

**Review by rw-access — APPROVED** [review]

**Pull request merged**